nni/experiment/config/utils/internal.py

```diff
@@ -198,3 +203,33 @@ def get_ipv4_address() -> str:
     addr = s.getsockname()[0]
     s.close()
     return addr
+
+def load_experiment_config(config_json: dict) -> ExperimentConfig | RetiariiExeConfig:
+    # avoid circular import
+    from nni.nas.experiment.config import RetiariiExeConfig
+    from nni.nas.experiment.pytorch import RetiariiExperiment
+    from ..experiment_config import ExperimentConfig
+    from ...experiment import Experiment
+    exp_cls = get_experiment_cls_using_config(config_json)
+    if exp_cls is Experiment:
+        return ExperimentConfig(**config_json)
+    elif exp_cls is RetiariiExperiment:
+        return RetiariiExeConfig(**config_json)
+    else:
+        raise TypeError(f'Unsupported experiment type: {type(exp_cls)}')
+
+def get_experiment_cls_using_config(config_json: dict) -> type[Experiment] | type[RetiariiExperiment]:
+    from nni.nas.experiment.pytorch import RetiariiExperiment
+    from ...experiment import Experiment
+    if 'experimentType' in config_json:
+        if config_json['experimentType'] == 'hpo':
+            return Experiment
+        elif config_json['experimentType'] == 'nas':
+            return RetiariiExperiment
```

Comment:
The `import` can be put above this line.

Reply:
updated